File: Assets/Scripts/ShortCut.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ShortCut : MonoBehaviour
{

	private static GameObject _engine;

	private GameObject GetEngine()
	{
		if (_engine == null) _engine = GameObject.FindWithTag("GameEngine");
		return _engine;
	}

	protected Assets Assets
	{
		get { return GetEngine().GetComponent<Assets>(); }
	}

	protected GameObject Canvas
	{
		get { return GameObject.Find("Canvas"); }
	}

	protected LuaState LuaState
	{
		get { return GetEngine().GetComponent<LuaState>(); }
	}

	protected Scenes Scenes
	{
		get { return GetEngine().GetComponent<Scenes>(); }
	}
}

File: Assets/Lua/shortcut.lua
local _M = { _engine = nil}

function _M:engine()
    if not _M._engine then
        _M._engine = CS.UnityEngine.GameObject.FindWithTag("GameEngine")
    end
    return _M._engine
end

function _M:Assets()
    return self:engine():GetComponent("Assets")        
end

function _M:Scenes()
    return self:engine():GetComponent("Scenes")        
end

function _M:Canvas()
    return CS.UnityEngine.GameObject.Find("Canvas")
end

return _M